main.cpp

```diff
@@ -70,75 +62,44 @@ i cubed: 531441
 Use a service like https://www.diffchecker.com/diff to compare your output. 
 */
 
-#include <typeinfo>
-#include <memory>
-#include<iostream>
-#include<cmath>
-#include <functional>
-
-template<typename NumericType>
-struct Temporary
-{
-    Temporary(NumericType t) : v(t)
-    {
-        std::cout << "I'm a Temporary<" << typeid(v).name() << "> object, #"
-                  << counter++ << std::endl;
-    }
-
-    operator NumericType() const 
-    { 
-        return v;
-    }
-    operator NumericType&() 
-    {
-       return v;
-    }
-private:
-    static int counter;
-    NumericType v;
-};
-
-
-template<typename Type>
-int Temporary<Type>::counter = 0;
-
-
-struct A {};
-struct HeapA
-{ 
-    HeapA() : a(new A) {}
-    ~HeapA()
-    {
-        delete a;
-    }
-    A* a = nullptr;
-};
-
 #include <iostream>
 #include <cmath>
 #include <functional>
 #include <memory>
 #include <typeinfo>
 #include <utility>
+#include "LeakedObjectDetector.h"
 
 template<typename NumericType>
 struct Temporary
 {
-    Temporary(NumericType t) : v(t)
+    Temporary(const NumericType& t) : v(t) { incCounter(); }
+    Temporary(NumericType&& t) : v(std::move(t)) { incCounter(); }
```

Comment:
This constructor does not need to exist. 
This is not one of the Rule-Of-5 constructors. 
- [ ] remove this constructor